src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -92,19 +96,25 @@ static MsQuicApi()
         if (!loaded)
         {
             // MsQuic library not loaded
+            NotSupportedReason = $"Unable to load MsQuic library version '{s_minMsQuicVersion.Major}'.";
```

Comment:
Should we put string allocations to under `if (NetEventSource.Log.IsEnabled())`?

Reply:
This needs to be available without logging as we use it in the NotSupportedException. But it happens only in case we fail to load MsQuic and it's one string per app.